packages/flutter/lib/src/material/material.dart

```diff
@@ -108,17 +108,19 @@ abstract class MaterialInkController {
 class Material extends StatefulWidget {
   /// Creates a piece of material.
   ///
-  /// The [type] and the [elevation] arguments must not be null.
+  /// The [type], [elevation] and [shadowColor] arguments must not be null.
   const Material({
     Key key,
     this.type: MaterialType.canvas,
     this.elevation: 0.0,
     this.color,
+    this.shadowColor: const Color(0xFF000000),
     this.textStyle,
     this.borderRadius,
     this.child,
   }) : assert(type != null),
        assert(elevation != null),
+       assert(shadowColor != null),
```

Comment:
`color` seems to be missing an assert here?

Reply:
We're not animating color in `Material`, so `null` might be a valid value, although I haven't checked myself.

Reply:
Yeah, we don't want to, but not animating it doesn't mean it could be null — material has a colour, be it white (default) or some other colour. I'm not sure what a null colour would represent for a material, semantically?

Reply:
Found it. Look for `_getBackgroundColor` in this file. `null` is a valid value for `color`, and means get the color from `ThemeData` based on the material type. We might do the same for `shadowColor` eventually, but for now let's just `assert`.